File: tests/fixtures/sample_repo/src/calculator.py
"""Simple calculator module for testing."""

from .utils import validate_number


def add(a, b):
    """Add two numbers."""
    validate_number(a)
    validate_number(b)
    return a + b


def subtract(a, b):
    """Subtract b from a."""
    validate_number(a)
    validate_number(b)
    return a - b


def multiply(a, b):
    """Multiply two numbers."""
    validate_number(a)
    validate_number(b)
    return a * b


def divide(a, b):
    """Divide a by b."""
    validate_number(a)
    validate_number(b)
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b


class Calculator:
    """Stateful calculator with history."""

    def __init__(self):
        self.history = []

    def compute(self, op, a, b):
        ops = {"add": add, "subtract": subtract, "multiply": multiply, "divide": divide}
        if op not in ops:
            raise ValueError(f"Unknown operation: {op}")
        result = ops[op](a, b)
        self.history.append((op, a, b, result))
        return result

    def last_result(self):
        if not self.history:
            return None
        return self.history[-1][3]

File: tests/fixtures/sample_java_repo/src/main/java/calculator/Calculator.java
package calculator;

import utils.Utils;

/**
 * Simple calculator with basic arithmetic operations.
 */
public class Calculator {

    private double lastResult;

    public Calculator() {
        this.lastResult = 0;
    }

    /**
     * Add two numbers.
     */
    public double add(double a, double b) {
        Utils.validateNumber(a);
        Utils.validateNumber(b);
        this.lastResult = a + b;
        return this.lastResult;
    }

    /**
     * Subtract b from a.
     */
    public double subtract(double a, double b) {
        this.lastResult = a - b;
        return this.lastResult;
    }

    public double getLastResult() {
        return this.lastResult;
    }
}
